### PR TITLE
Fix delegation.

### PR DESCRIPTION
**Pull request by dlongley**
This PR fixes non-root delegation and adds the required LD signature suite param to the constructor. While this param is only needed for delegation at the moment, I expect it will be needed for http signatures to provide algorithm information once http signatures is updated. We may end up having a separate property for it, but I'm going to make it required for now to keep future updates simpler.

### CODE REVIEW

**Comment by dlongley** [comment]
Also, there is some readme template file (`.hbs`) and the actual `README.md` here. That's not typical for our libs, so I don't know what's going on there, I presume the former is supposed to generate the latter or something? I'm not touching that right now, don't have time for it. So I just updated both in this PR.

File: lib/ZcapClient.js [hunk 1]
```diff
@@ -28,48 +41,77 @@ export class ZcapClient {
...
+   * @param {Function} [options.documentLoader] - Optional document loader
+   *   to load suite-related contexts. If none is provided, one will be
+   *   auto-generated if the suite class expresses its required context.
    *
    * @returns {ZcapClient} - The new ZcapClient instance.
    */
   constructor({
     baseUrl, defaultHeaders = {}, agent, didDocument, keyPairs,
-    invocationSigner, delegationSigner
+    invocationSigner, delegationSigner, SuiteClass, documentLoader
```
Comment (msporny) [inline lib/ZcapClient.js:74]: Are options supposed to be in alpha order? or mandatory/optional alpha?

**Review by msporny — APPROVED** [review]
Approving, see question. I fixed the README generation. 'npm run generate-readme' ... it's non-standard, but I wanted us to have decent documentation around this library since it was going to be community-facing.

File: lib/ZcapClient.js (hunk 1 above)
Comment (dlongley) [reply lib/ZcapClient.js:74]: mandatory/optional alpha -- and I'll reorder this.

**Pull request merged**